File: Practice11_1/app/src/main/java/juicyjung/gachon/practice11_1/MainActivity.java
package juicyjung.gachon.practice11_1;

import androidx.appcompat.app.AppCompatActivity;

import android.os.AsyncTask;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    TextView tvCounter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        tvCounter = (TextView) findViewById(R.id.tv_counter);
        new CountDownTask().execute(); //Starts the CountDownTask
    }

    private class CountDownTask extends AsyncTask<Void, Integer, Void> {
        @Override
        protected void onPreExecute() {
            tvCounter.setText("*START*");
        }

        @Override
        protected Void doInBackground(Void... params) {
            for (int i = 15; i >= 0; i--) {
                try {
                    Thread.sleep(1000);
                    publishProgress(i); //Invokes onProgressUpdate();
                } catch (Exception e) {
                }
            }
            return null;
        }

        @Override
        protected void onProgressUpdate(Integer... values) {
            tvCounter.setText(Integer.toString(values[0].intValue()));
        }

        @Override
        protected void onPostExecute(Void aVoid) {
            tvCounter.setText("*DONE*");
        }
    }
}

File: Practice9_1/app/src/main/java/juicyjung/gachon/practice9_1/MainActivity.kt
package juicyjung.gachon.practice9_1

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import juicyjung.gachon.practice9_1.databinding.ActivityMainBinding

private lateinit var binding: ActivityMainBinding

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        var count: Int = 0

        binding.textview.text = count.toString()

        binding.btn.setOnClickListener{
            count++
            binding.textview.text = count.toString()
        }
    }
}